incubator/binding-asyncapi/src/main/java/io/aklivity/zilla/runtime/binding/asyncapi/internal/AsyncapiCompositeBindingAdapter.java

```diff
@@ -90,6 +95,21 @@ protected AsyncapiProtocol resolveProtocol(
         return protocol;
     }
 
+    protected void resolveServerVariables(
+        Asyncapi asyncApi)
+    {
+        for (AsyncapiServer s : asyncApi.servers.values())
+        {
+            String[] hostAndPort = s.host.split(":");
+            String port = hostAndPort[1];
+            if (variable.reset(port).find())
+            {
+                String resolvedPort = s.variables.get(variable.group(1)).defaultValue;
+                s.host = s.host.replace(port, resolvedPort);
+            }
+        }
```

Comment:
```suggestion
            s.host = variable.reset(s.host).replaceAll(mr -> s.variables.get(mr.group(1)).defaultValue);
        }
```
This should handle variables for host and/or port.